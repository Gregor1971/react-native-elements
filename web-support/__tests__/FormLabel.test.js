import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../samples';
import { FormLabel as Component } from '../../src';

const props = {
  containerStyle: {
    component: Component,
    children: 'enter something',
    props: { containerStyle: { backgroundColor: '#2ff', padding: 5 } },
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
  labelStyle: {
    component: Component,
    children: 'enter something',
    props: { labelStyle: { backgroundColor: '#2ff', padding: 5 } },
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
  fontFamily: {
    component: Component,
    children: 'enter something',
    props: { fontFamily: 'Courier New' },
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Forms: { sectionComponents: { FormLabel: { samples } } },
  },
  options
);
