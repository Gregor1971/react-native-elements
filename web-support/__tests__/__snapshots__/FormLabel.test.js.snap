// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forms: FormLabel: [ prop: containerStyle ] enzyme-mount: snapshot 1`] = `
<FormLabel
  containerStyle={
    Object {
      "backgroundColor": "#2ff",
      "padding": 5,
    }
  }
>
  <View
    style={
      Array [
        142,
        Object {
          "backgroundColor": "#2ff",
          "padding": 5,
        },
      ]
    }
  >
    <div
      className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingBottom-1inuy60 rn-position-bnwqim"
      style={
        Object {
          "backgroundColor": "rgba(34,255,255,1)",
          "paddingLeft": "5px",
          "paddingRight": "5px",
          "paddingTop": "5px",
        }
      }
    >
      <TextElement
        style={
          Array [
            143,
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              64,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                143,
                undefined,
                undefined,
              ],
            ]
          }
        >
          <div
            className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-zzr6i3 rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1enofrn rn-fontWeight-vw2c0b rn-marginBottom-10sqg0u rn-marginLeft-1joea0r rn-marginRight-1mf7evn rn-marginTop-19h5ruw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
            dir="auto"
          >
            enter something
          </div>
        </Text>
      </TextElement>
    </div>
  </View>
</FormLabel>
`;

exports[`Forms: FormLabel: [ prop: containerStyle ] enzyme-render: snapshot 1`] = `
<div
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingBottom-1inuy60 rn-position-bnwqim"
  style="background-color:rgba(34,255,255,1);padding-top:5px;padding-right:5px;padding-left:5px;"
>
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-zzr6i3 rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1enofrn rn-fontWeight-vw2c0b rn-marginBottom-10sqg0u rn-marginLeft-1joea0r rn-marginRight-1mf7evn rn-marginTop-19h5ruw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
  >
    enter something
  </div>
</div>
`;

exports[`Forms: FormLabel: [ prop: containerStyle ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      142,
      Object {
        "backgroundColor": "#2ff",
        "padding": 5,
      },
    ]
  }
>
  <TextElement
    style={
      Array [
        143,
        undefined,
        undefined,
      ]
    }
  >
    enter something
  </TextElement>
</View>
`;

exports[`Forms: FormLabel: [ prop: fontFamily ] enzyme-mount: snapshot 1`] = `
<FormLabel
  fontFamily="Courier New"
>
  <View
    style={
      Array [
        142,
        undefined,
      ]
    }
  >
    <div
      className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
    >
      <TextElement
        style={
          Array [
            143,
            undefined,
            Object {
              "fontFamily": "Courier New",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              64,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                143,
                undefined,
                Object {
                  "fontFamily": "Courier New",
                },
              ],
            ]
          }
        >
          <div
            className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-zzr6i3 rn-display-1471scf rn-font-1lw9tu2 rn-fontSize-1enofrn rn-fontWeight-vw2c0b rn-marginBottom-10sqg0u rn-marginLeft-1joea0r rn-marginRight-1mf7evn rn-marginTop-19h5ruw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
            dir="auto"
            style={
              Object {
                "fontFamily": "Courier New",
              }
            }
          >
            enter something
          </div>
        </Text>
      </TextElement>
    </div>
  </View>
</FormLabel>
`;

exports[`Forms: FormLabel: [ prop: fontFamily ] enzyme-render: snapshot 1`] = `
<div
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
>
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-zzr6i3 rn-display-1471scf rn-font-1lw9tu2 rn-fontSize-1enofrn rn-fontWeight-vw2c0b rn-marginBottom-10sqg0u rn-marginLeft-1joea0r rn-marginRight-1mf7evn rn-marginTop-19h5ruw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
    style="font-family:Courier New;"
  >
    enter something
  </div>
</div>
`;

exports[`Forms: FormLabel: [ prop: fontFamily ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      142,
      undefined,
    ]
  }
>
  <TextElement
    style={
      Array [
        143,
        undefined,
        Object {
          "fontFamily": "Courier New",
        },
      ]
    }
  >
    enter something
  </TextElement>
</View>
`;

exports[`Forms: FormLabel: [ prop: labelStyle ] enzyme-mount: snapshot 1`] = `
<FormLabel
  labelStyle={
    Object {
      "backgroundColor": "#2ff",
      "padding": 5,
    }
  }
>
  <View
    style={
      Array [
        142,
        undefined,
      ]
    }
  >
    <div
      className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
    >
      <TextElement
        style={
          Array [
            143,
            Object {
              "backgroundColor": "#2ff",
              "padding": 5,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              64,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                143,
                Object {
                  "backgroundColor": "#2ff",
                  "padding": 5,
                },
                undefined,
              ],
            ]
          }
        >
          <div
            className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-zzr6i3 rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1enofrn rn-fontWeight-vw2c0b rn-marginBottom-10sqg0u rn-marginLeft-1joea0r rn-marginRight-1mf7evn rn-marginTop-19h5ruw rn-paddingBottom-1inuy60 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
            dir="auto"
            style={
              Object {
                "backgroundColor": "rgba(34,255,255,1)",
                "paddingLeft": "5px",
                "paddingRight": "5px",
                "paddingTop": "5px",
              }
            }
          >
            enter something
          </div>
        </Text>
      </TextElement>
    </div>
  </View>
</FormLabel>
`;

exports[`Forms: FormLabel: [ prop: labelStyle ] enzyme-render: snapshot 1`] = `
<div
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
>
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-zzr6i3 rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1enofrn rn-fontWeight-vw2c0b rn-marginBottom-10sqg0u rn-marginLeft-1joea0r rn-marginRight-1mf7evn rn-marginTop-19h5ruw rn-paddingBottom-1inuy60 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
    style="background-color:rgba(34,255,255,1);padding-top:5px;padding-right:5px;padding-left:5px;"
  >
    enter something
  </div>
</div>
`;

exports[`Forms: FormLabel: [ prop: labelStyle ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      142,
      undefined,
    ]
  }
>
  <TextElement
    style={
      Array [
        143,
        Object {
          "backgroundColor": "#2ff",
          "padding": 5,
        },
        undefined,
      ]
    }
  >
    enter something
  </TextElement>
</View>
`;
